implementations/rust/ockam/ockam_api/src/ui/output/mod.rs

```diff
@@ -37,10 +39,34 @@ use std::fmt::Write;
 /// }
 /// ```
 pub trait Output {
+    /// Format to use when the item is printed as a standalone item
     fn item(&self) -> Result<String>;
 
+    /// Format to use when the item is part of a list.
+    /// By default, the list representation is the same as the standalone representation
+    /// but removing the padding of each line.
     fn as_list_item(&self) -> Result<String> {
-        self.item()
+        Ok(self
+            .item()?
+            .lines()
+            .map(|l| l.strip_prefix(fmt::PADDING).unwrap_or(l))
+            .join("\n"))
+    }
+
+    /// Adds padding to each line of the Display output
+    fn padded_display(&self) -> std::result::Result<String, Error>
+    where
+        Self: Display,
+    {
+        Ok(self.iter_output()?.pad().to_string())
+    }
+
+    /// Returns an iterator over the lines of the Display output
+    fn iter_output(&self) -> std::result::Result<OutputIter, Error>
```

Comment:
There are two things that I find a bit troubling:

 - The fact that we need to declare that output functions can throw errors. It seems to me that as long as I have correct data, I should always be able to display it.
 
 - The fact that we require different error types for different methods of the trait. It looks like this is due to the fact that we're trying to reuse some `Output` code to implement some `Display` instance. I think we should consider one being lower level than the other and use only code in one direction (like `Display` inside `Output` but not the opposite).

Reply:
> The fact that we need to declare that output functions can throw errors. It seems to me that as long as I have correct data, I should always be able to display it.

There could be cases where we need to do some data transformation to build the output. I'm not sure if have a case like that, but I didn't want to close that door when I first introduced this trait.

> The fact that we require different error types for different methods of the trait....

Right, those functions don't need to return a result. It's a leftover from a previous prototype.

Reply:
> There could be cases where we need to do some data transformation to build the output. I'm not sure if we have a case like that, but I didn't want to close that door when I first introduced this trait.

I still don't think that this should not be an error. I think I'd rather see a partial output than getting an Error 500 because a field can not be transformed into the ideal format.